### ISSUE TITLE
Allow search pages to filter JournalEntryPage#toc when "Full Text Search" is enabled

### ISSUE THREAD

**Issue by tdakanalis** [farling42/fvtt-pdf-pager#93]
Although PDF pages are displayed as a `JournalEntryPage#toc`, the `Search Pages` field does not allow you to filter them.

**Comment by farling42** [comment]
I'll investigate this, whilst adding lots of notes to this issue about the code that might need changing.

**Comment by farling42** [comment]
With the magnifying glass icon present next to the "Search Pages" box, the search is only performed based on the Journal Page title (not any sections within the PDF).

With the "Full Text Search" icon present, no text is found.

**Comment by farling42** [comment]
For searching within a PDF, the best option is to use the search facility within the PDF viewer itself. This can be opened by pressing Ctrl-F.

**Comment by tdakanalis** [comment]
Actually I was referring to searching among the titles/names not a full text search
[Screencast from 2024-03-22 14-27-21.webm](https://github.com/farling42/fvtt-pdf-pager/assets/2813615/bb12ba7a-884b-425e-a1bb-5e7c9af7a294)

**Comment by tdakanalis** [comment]
If you could index the PDF sections as Journal Entry Pages this could work but I am not sure if it is possible.

**Comment by farling42** [comment]
Filtering on the left only occurs on Journal Page titles, not individual sections (as far as I can tell from testing on a normal text-based journal with multiple pages and sections within each page).

Unfortunately, a PDF exists as a single journal page within an overall journal entry. It is possible to store multiple PDFs in a single journal entry (each as a separate page), or mixed in with pages of other types.

**Comment by tdakanalis** [comment]
Yes, I understand that, but if you could create an in-memory JournalEntryPage tree, each one pointing to the respective section `@PDF[Core Rules|making-rolls]{Making Rolls}`  then it probably could work. You can trigger that the first time someone opens a PDF and then keep the structure in memory so that also other modules like the `Quick Insert - Search Widget` could also index and search among them.

**Comment by farling42** [comment]
I will have a look at that module and see where it is getting it's search information from, to see if it is feasible to index the TOC of PDFs.

**Comment by farling42** [comment]
See https://gitlab.com/fvtt-modules-lab/quick-insert/-/issues/125 which requests that `Quick Insert - Search Widget` supports PDF Pager.

**Comment by farling42** [comment]
This functionality needs to be added to Quick Insert (see linked issue above) or another module. Thus I am closing the issue on the PDF Pager module.
